BrainPortal/app/views/users/new.html.erb

```diff
@@ -76,6 +76,8 @@
     <p><label for="no_password_reset_needed">No need to reset initial password:</label>
     <%= check_box_tag :no_password_reset_needed, "1", params[:no_password_reset_needed] == "1" %>
 
+    <%= hidden_field_tag :signup_id, params[:signup_id] %>
```

Comment:
If the form fails to validate, the new() action is not invoked at all, and we don't need to signup_id anymore, it was already used to populate all the fields. It IS however needed in create() when making the association back to the signup models.